### DISCUSSION TITLE
Time trial

### DISCUSSION THREAD

**Discussion by chillfox** [BattlesnakeOfficial/feedback#224] [General]
This is a bit like timed chess.

So your snake gets 10 minutes to play the game, there are no per-turn timeouts. The time you use each turn is subtracted from the total time and when you are out you just keep going in whatever direction you went last with the error message "out of time". 

So you can blow the entire 10 min in the first turn to make a plan or try and be very fast early on, then midgame take a multi-minute turn to compute a plan that you just follow for the remaining turns.

**Comment by robbles** [comment]
I think this is such a cool idea! The strategies around managing remaining time would be really interesting to see.

How do you see time being exposed to snakes? I wonder if replacing the health field with a time remaining in milliseconds would work? That way (in theory, maybe with some code changes) the board could also show the remaining time.

**Reply by chillfox** [reply]
My first thought was to just use the existing timeout property, but yeah health could work as well.
oh, yeah timeout won't really work because you want to see how much the other snakes have left.

So making it a game mode where time is health and food only provides length (or maybe an extra 50 ms of life)...
I think this has a lot of potential to be really fun.
